reps/2023-04-28-remove-algorithms-from-rllib.md

```diff
@@ -0,0 +1,193 @@
+
+if __name__ == "__main__":
+    ray.init()
+    register_env(
+        "cartpole",
+        lambda env_cfg: TimeLimit(CartPoleMassEnv(), max_episode_steps=200),
+    )
+
+    config = MAMLConfig()
+
+    tuner = tune.Tuner(
+        MAML,
+        param_space=config.to_dict(),
+        run_config=air.RunConfig(stop={"training_iteration": 100}),
+    )
+    results = tuner.fit()
+
+```
+
+
+This proposed algorithms to remove are:
+- A3C
+- AphaStar
+- AlphaZero
+- ApexDQN
+- ApexDDPG
+- DDPG
+- ARS
+- Bandits 
+- CRR 
+- DDPG
+- DDPPO
+- Dreamer
+- DT
+- ES
+- Leelachess
+- MADDPG
+- MAML
+- PG
+- QMix
+- Random
+- SimpleQ
+- SlateQ
+- TD3
+- R2D2
+
+
+### Shepherd of the Proposal (should be a senior committer)
+
+@sven1977
+
+## Design and Architecture
+
+- We introduce the new  repo for deprecated RLlib algorithms and community contributed algorithms.
+
+### `rayproject/rllib-contrib` File Structure
+
+```
+README.md
+.github/
+    workflow/
+        algorithm_foo.yml <---------- This will launch the unit test for algorithm foo with gh actions
+algorithm_foo/
+    src/
+        algorithm_foo.py
+    tests/
+    examples/
+    requirements.txt
+    pyproject.toml  <---------- Used for installation
+
+latest_results/
+    algorithm_foo/
+        results.json
+        results.csv
+        results.pkl
+        results.tensorboard
+        results.md  <---------- This will contain some pictures of the performance as a sanity check
+```
+
+### Installation
+1. use git to clone `rayproject/rllib-contrib`
+2. navigate to the directory of the previously deprecated algorithm
+   for ex. `cd rllib-contrib/maml`
+3. run `pip install -e .` to install the algorithm as a pip package
+4. Import the algorithm, its policies and its config from that package instead of from `ray.rllib`
+
+
+## Contribution Guidlines
+
+
+The RLlib team commits to the following level of support for the algorithms in this repo:
+
+| Platform | Purpose | Estimated Response Time | Support Level |
+| --- | --- | --- | --- |
+| [Discuss Forum](https://discuss.ray.io) | For discussions about development and questions about usage. | < 1 week | Community |
+| [GitHub Issues](https://github.com/ray-project/rllib-contrib-maml/issues) | For reporting bugs and filing feature requests. | N/A | Community |
+| [Slack](https://forms.gle/9TSdDYUgxYs8SA9e8) | For collaborating with other Ray users. | < 2 weeks | Community |
+
+**This means that any issues that are filed will be solved best-effort by the community and there is no expectation of maintenance by the RLlib team.**
+
+We will generally accept contributions to this repo that meet any of the following criteria:
+1. Updating dependencies.
+2. Submitting community contributed algorithms that have been tested and are ready for use.
+3. Enabling algorithms to be run in different environments (ex. adding support for a new type of gym environment).
+4. Updating algorithms for use with the newer RLlib API.
```

Comment:
I think we should advertise this as great community projects!

There is a clear path for member of the community to migrate algorithms to new APIs when looking at how other algorithms have been migrated and it makes for a great practice to learn RLlib's inner APIs.
For example for RLModules. If someone would write a clean DQN RLModule, they could contribute it. Stuff like that.

Reply:
You also might want to write some guidelines how the new APIs should be used when writing - sometimes complex - algorithms.